tests/integration-service/status-reporting-to-pullrequest.go

```diff
@@ -0,0 +1,234 @@
+			})
+		})
+
+		When("Integration PipelineRuns completes successfully", func() {
+			It("should lead to Snapshot CR being marked as failed", FlakeAttempts(3), func() {
+				// Snapshot marked as Failed because one of its Integration test failed (as expected)
+				Eventually(func() bool {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot("", pipelineRun.Name, "", testNamespace)
+					return err == nil && !f.AsKubeAdmin.CommonController.HaveTestsSucceeded(snapshot)
+				}, time.Minute*3, time.Second*5).Should(BeTrue(), fmt.Sprintf("Timed out waiting for Snapshot to be marked as failed %s/%s", snapshot.GetNamespace(), snapshot.GetName()))
+			})
+			It("eventually leads to the status reported at Checks tab for the successful Integration PipelineRun", func() {
+				validateCheckRun(*f, integrationTestScenarioPass.Name, checkrunConclusionSuccess, helloWorldComponentGitSourceRepoName, prHeadSha, prNumber)
+			})
+			It("eventually leads to the status reported at Checks tab for the failed Integration PipelineRun", func() {
+				validateCheckRun(*f, integrationTestScenarioFail.Name, checkrunConclusionFailure, helloWorldComponentGitSourceRepoName, prHeadSha, prNumber)
+			})
+		})
+	})
+})
+
+// validateCheckRun fetches a specific CheckRun within a given repo
+// by matching the CheckRun's name with the given checkRunName, and
+// then validates that it got completed and its conclusion is as expected
+func validateCheckRun(f framework.Framework, checkRunName, conclusion, repoName, prHeadSha string, prNumber int) {
+	var checkRun *github.CheckRun
+	var timeout, interval time.Duration
+
+	timeout = time.Minute * 10
+	interval = time.Second * 2
+
+	Eventually(func() *github.CheckRun {
+		checkRuns, err := f.AsKubeAdmin.CommonController.Github.ListCheckRuns(repoName, prHeadSha)
+		Expect(err).ShouldNot(HaveOccurred())
+		for _, cr := range checkRuns {
+			if strings.Contains(cr.GetName(), checkRunName) {
+				checkRun = cr
+				return cr
+			}
+		}
+		return nil
+	}, timeout, interval).ShouldNot(BeNil(), fmt.Sprintf("timed out when waiting for the PaC CheckRun, with `Name` field containing the substring %s, to appear in the PR #%d of the Component repo %s", checkRunName, prNumber, repoName))
+
+	Eventually(func() string {
+		checkRun, err := f.AsKubeAdmin.CommonController.Github.GetCheckRun(repoName, checkRun.GetID())
+		Expect(err).ShouldNot(HaveOccurred())
+		return checkRun.GetStatus()
+	}, timeout, interval).Should(Equal(checkrunStatusCompleted), fmt.Sprintf("timed out when waiting for the PaC Check suite status to be 'completed' in the Component repo %s in PR #%d", repoName, prNumber))
+	Expect(checkRun.GetConclusion()).To(Equal(conclusion), fmt.Sprintf("the PR %d in %s repo doesn't contain the expected conclusion (%s) of the CheckRun", prNumber, repoName, conclusion))
+}
```

Comment:
nice work!

just thinking if we should update docs [here](https://github.com/redhat-appstudio/e2e-tests/blob/main/tests/integration-service/README.md) and mention also [here](https://github.com/redhat-appstudio/e2e-tests/blob/main/docs/Installation.md#advanced-installation-guidelines), that Github App is required for this test, wdyt?

Reply:
Oh, I totally forgot about those places, will update it now. Thanks!

Reply:
updated, PTAL